### PR TITLE
Prepare release v0.42.0

### PR DESCRIPTION
**Pull request by srenatus**
🎉

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,7 +3,139 @@
...
+
+Note that one-expression bodies can be abbreviated when using `if`.
+
+To use the new keywords, use `import future.keywords.contains` and `import future.keywords.if`; or
+import all of them at once via `import future.keywords`. When these future imports are present, the
+pretty printer (`opa fmt`) will introduce `contains` and `if` where applicable.
+
+`if` is allowed in all places to separe the rule head from the body, like
+```rego
+foo[x] = y if { x := "a", y := "b" }
```
Comment (anderseknert) [inline CHANGELOG.md:48]: Could we try and keep up with the “real world” policy example, ie avoid using “foo” and so on?

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,7 +3,139 @@
...
+- prometheus: bump client_golang, adjust setup and tests (#4755) ([#4697](https://github.com/open-policy-agent/opa/issues/4697))
+- Add e2e mode for opa bench
+- SDK: Allow configurable ID (#4736) (authored by @rakshasa-1729)
+
+### Rego and Topdown
+- Fix wrong runtime type error message in count, object.filter, object.remove (#4768) ([#4767](https://github.com/open-policy-agent/opa/issues/4767))
+
+- ast/parser: Remove early MHS return in infix parsing. (#4773) ([#4672](https://github.com/open-policy-agent/opa/issues/4672)) authored by @philipaconrad
+- ast: Disallow shadowing of called functions in comprehension heads (#4701) ([#4762](https://github.com/open-policy-agent/opa/issues/4762))
+- built-ins: add object.subset() builtin ([#4358](https://github.com/open-policy-agent/opa/issues/4358)) authored by @charlesdaniels
```
Comment (anderseknert) [inline CHANGELOG.md:71]: I think we could highlight this better.

**Review by anderseknert — COMMENTED** [review]
Looks great! Just some suggestions, but except for the typo nothing that needs to be changed.

File: CHANGELOG.md (hunk 1 above)
Comment (srenatus) [reply CHANGELOG.md:48]: :+1: good point

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,7 +3,135 @@
...
+We still use our release notes to record the nice fixed contributed by our community.
+
+- Ecosystem Additions:
+  - Alfred, the self-hosted playground (authored by @dolevf)
+  - Java Spring tutorial (authored by @psevestre)
+  - Pulumi
+
+### Miscellaneous
+
+- ci: bump golangci-lint to v1.46.2, fix some issues (#4809) ([#4765](https://github.com/open-policy-agent/opa/issues/4765)) reported by @philipaconrad
```
Comment (srenatus) [inline CHANGELOG.md:122]: this block also isn't edited yet

**Pull request merged**